components/configs/rwn-optimizations/base/3-node/worker-fix-ipi-rwn.yaml

```diff
@@ -0,0 +1,46 @@
+---
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  name: 50-worker-fix-ipi-rwn
+  labels:
+    machineconfiguration.openshift.io/role: worker
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    storage:
+      files:
+        - contents:
+            source: data:,
+            verification: {}
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/keepalived.yaml
+        - contents:
+            source: data:,
+            verification: {}
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/mdns-publisher.yaml
+        - contents:
+            source: data:,
+            verification: {}
+          filesystem: root
+          mode: 420
+          path: /etc/kubernetes/manifests/coredns.yaml
+    systemd:
+      units:    # yamllint disable rule:line-length
+        - contents: |
+            [Unit]
+            Description=Writes IP address configuration so that kubelet and crio services select a valid node IP
+            Wants=network-online.target
+            After=network-online.target ignition-firstboot-complete.service
+            Before=kubelet.service crio.service
+            [Service]
+            Type=oneshot
+            ExecStart=/bin/bash -c "exit 0 "
+            [Install]
+            WantedBy=multi-user.target
+          enabled: true   # yamllint disable rule:line-length
+          name: nodeip-configuration.service
```

Comment:
I really don't think this is needed any longer, we should really test this.

Reply:
That is a doubt I had. I remember this was discovered when doing an upgrade in OCP 4.10. Only needed for upgrades in 4.10.x then?

Reply:
@novacain1 Looking good in OCP 4.11 and 4.12. No need of that file.
```
[Unit]
Description=Writes IP address configuration so that kubelet and crio services select a valid node IP
Wants=NetworkManager-wait-online.service crio-wipe.service
After=NetworkManager-wait-online.service ignition-firstboot-complete.service crio-wipe.service
Before=kubelet.service crio.service ovs-configuration.service

[Service]
# Need oneshot to delay kubelet
Type=oneshot
# Would prefer to do Restart=on-failure instead of this bash retry loop, but
# the version of systemd we have right now doesn't support it. It should be
# available in systemd v244 and higher.
ExecStart=/bin/bash -c " \
  until \
  /usr/bin/podman run --rm \
  --authfile /var/lib/kubelet/config.json \
  --net=host \
  --security-opt label=disable \
  --volume /etc/systemd/system:/etc/systemd/system \
  --volume /run/nodeip-configuration:/run/nodeip-configuration \
  quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:6b61429030b790e6ec6e9fcb52b2a17c5b794815d6da9806bc563bc45e84aa67 \
  node-ip \
  set \
  --retry-on-failure \
  ${NODEIP_HINT:-${KUBELET_NODEIP_HINT:-}}; \
  do \
  sleep 5; \
  done"
ExecStart=/bin/systemctl daemon-reload
ExecStartPre=/bin/mkdir -p /run/nodeip-configuration

EnvironmentFile=-/etc/default/nodeip-configuration

[Install]
RequiredBy=kubelet.service
```